dev/django-admin.ts

```diff
@@ -0,0 +1,1290 @@
+        description: "Skip system checks.",
+      },
+    ],
+    description: "Creates new migration(s) for apps.",
+    args: [
+      {
+        name: "app_label",
+        description: "Specify the app label(s) to create migrations for.",
+        isOptional: true,
+        variadic: true,
+      },
+    ],
+  },
+  {
+    name: "migrate",
+    icon: DJANGO_ICON_URL,
+    options: [
+      {
+        name: ["--noinput", "--no-input"],
+        description:
+          "Tells Django to NOT prompt the user for input of any kind.",
+      },
+      {
+        name: "--database",
+        description:
+          'Nominates a database to synchronize. Defaults to the "default" database.',
+        args: {
+          name: "DATABASE",
+        },
+      },
+      {
+        name: "--fake",
+        description: "Mark migrations as run without actually running them.",
+      },
+      {
+        name: "--fake-initial",
+        description:
+          "Detect if tables already exist and fake-apply initial migrations if so. Make sure that the current database schema matches your initial migration before using this flag. Django will only check for an existing table name.",
+      },
+      {
+        name: "--plan",
+        description:
+          "Shows a list of the migration actions that will be performed.",
+      },
+      {
+        name: "--run-syncdb",
+        description: "Creates tables for apps without migrations.",
+      },
+      {
+        name: "--check",
+        description:
+          "Exits with a non-zero status if unapplied migrations exist.",
+      },
+      {
+        name: "--skip-checks",
+        description: "Skip system checks.",
+      },
+    ],
+    description:
+      "Updates database schema. Manages both apps with migrations and those without.",
+    args: [
+      {
+        name: "app_label",
+        description: "App label of an application to synchronize the state.",
+        isOptional: true,
+      },
+      {
+        name: "migration_name",
+        description:
+          'Database state will be brought to the state after that migration. Use the name "zero" to unapply all migrations.',
+        isOptional: true,
+      },
+    ],
+  },
+  {
+    name: "sendtestemail",
+    icon: DJANGO_ICON_URL,
+    options: [
+      {
+        name: "--managers",
+        description:
+          "Send a test email to the addresses specified in settings.MANAGERS.",
+      },
+      {
+        name: "--admins",
+        description:
+          "Send a test email to the addresses specified in settings.ADMINS.",
+      },
+      {
+        name: "--skip-checks",
+        description: "Skip system checks.",
+      },
+    ],
+    description:
+      "Sends a test email to the email addresses specified as arguments.",
+    args: [
+      {
+        name: "email",
+        description: "One or more email addresses to send a test email to.",
+        isOptional: true,
+        variadic: true,
+      },
```

Comment:
Since this is one or more emails, I don't think it is optional.

```suggestion
      {
        name: "email",
        description: "One or more email addresses to send a test email to.",
        variadic: true,
      },
```

Reply:
Just checked this out 👀 

As I understand it, you're right that it's not optional UNLESS you pass the --managers or --admin options:
```bash
manage.py sendtestemail: error: You must specify some email recipients, or pass the --managers or --admin options.
```

So does that mean it should stay as `isOptional: true`? Or how can this sort of dynamic be best expressed in a Fig spec?

Reply:
This could be done using `generateSpec` (see https://fig.io/docs/reference/subcommand#generatespec). This function gives a variable named `context`, which is an array of the previous subcommands/options/args. We could then check if this array includes `managers` or `admin`, and return the corresponding subcommand.

Not sure if my explanation was clear enough - tell me if it wasn't the case!

Reply:
Ah I see @wilhelmklopp! Thanks for clarifying. I would add a comment here that makes a note of this and keep your original code.

@QuiiBz is right that you could use the under-documented `generateSpec` function on `sendtestemail ` to dynamically configure the value of `isOptional`.  My understanding is that `generateSpec` is somewhat expensive, so I'm not sure if it's worth running it for such a small change. cc @brendanfalk

Reply:
> I would add a comment here that makes a note of this and keep your original code.

Was just about to write this comment, but then GitHub Copilot started trolling me with its suggestion hahaha 😆 

![image](https://user-images.githubusercontent.com/7718702/125503224-cfe1b77c-4a16-40cc-af5d-c4d602bb9d27.png)

Reply:
Comment added in ce51309 ✅